S2VX.Game/Editor/ToolState/SelectToolState.cs

```diff
@@ -35,26 +36,93 @@ public class SelectToolState : S2VXToolState {
             return mouseInXRange && mouseInYRange;
         }
 
+        // https://gamedev.stackexchange.com/a/86784
+        private static Vector2 GetRotatedPoint(Vector2 point, Vector2 center, float rotation) {
+            var translatedX = point.X - center.X;
+            var translatedY = point.Y - center.Y;
+
+            var rotatedX = translatedX * (float)Math.Cos(rotation) - translatedY * (float)Math.Sin(rotation);
```

Comment:
Rather than converting to float after each operation, I think it'd make more sense to do all calculations with higher precision and then convert to float at the very end to avoid compounding any rounding errors in between.